File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // sorting-> time: O(nlogn), space: O(1)
        // store all numbers in the set, check for the longest inc sequence from that number

        // for each number, I check if num-1 exists, then skip, otherwise find longest from this number, time-> O(N), O(N)
        
        unordered_set<int>set1;
        int maxCons = 0;
        for(int num : nums){
            set1.insert(num);
        }
        for(int num : set1){
            if(set1.find(num-1)!=set1.end()){
                continue;
            }
            else{
                int cons = 1;
                while(set1.find(num+1)!=set1.end())
                {
                    cons++;
                    num++;
                }
                maxCons = max(cons, maxCons);
            }
        }
        return maxCons;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums.sort()
        if len(nums) == 0:
            return 0
        i = 1
        cons = 1
        maxCons = 1
        while i < len(nums):
            if nums[i] == nums[i-1]+1:
                print(nums[i])
                cons+=1
                print(cons)
            elif nums[i] == nums[i-1]:
                i+=1
                continue
            elif nums[i]!= nums[i-1]+1:
                cons = 1
            maxCons = max(cons, maxCons)
            i+=1
        return maxCons
            
